src/lib.rs

```diff
@@ -650,7 +651,7 @@ impl BlockStore {
                     let links = links(&block)?
                         .iter()
                         .map(CidBytes::try_from)
-                        .collect::<std::result::Result<Vec<_>, cid::Error>>()?;
+                        .collect::<std::result::Result<BTreeSet<_>, cid::Error>>()?;
```

Comment:
I think a FnvHashSet would be faster. I use it pretty much everywhere, so I don't mind the additional dependency, in case we don't have it already.